client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -372,6 +372,15 @@ private int calculateNumSegmentsToAcquire(ReaderGroupState state) {
     static Duration calculateAcquireTime(String readerId, ReaderGroupState state) {
         return TIME_UNIT.multipliedBy(state.getNumberOfReaders() - state.getRanking(readerId));
     }
+
+    void checkAndJoinStreamCutBarrier(Map<Segment, Long> ownedSegmentsWithOffset) throws ReinitializationRequiredException {
+        fetchUpdatesIfNeeded();
+        ReaderGroupState state = sync.getState();
+        String streamCutBarrierId = state.getBarrierIdForReader(readerId);  // Fetch the current StreamCutBarrier id.
```

Comment:
I think this is indicating that there can be only a single stream cut ongoing, is it right?